### PR TITLE
fix: Ensure that Drop API is called only when tx is successful

### LINKED ISSUE #9399: Ensure that Drop API is called only when tx is successful
> This is on database. all of them claimed = 1. but only 8 and 9 showup in the collection.

As I wrote on issue before - UI HAS TO ENSURE ThAT USER has successfully made the transaction.

_Originally posted by @vikiival in https://github.com/kodadot/private-workers/issues/64#issuecomment-1943810099_

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Bugfix

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #9399

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [x] My fix has changed UI

Mock tx error
Kapture 2024-02-15 at 14 34 53

  ## Copilot Summary
  copilot:summary

  copilot:poem

### CODE REVIEW

File: components/collection/drop/HolderOfGenerative.vue [hunk 1]
```diff
@@ -293,6 +295,11 @@ const mintNft = async () => {
 
 watch(status, (curStatus) => {
   if (curStatus === TransactionStatus.Block) {
+    if (isTransactionError.value) {
```
Comment (preschian) [inline components/collection/drop/HolderOfGenerative.vue:298]: add this to paid generative also?

**Review by preschian — APPROVED** [review]

**Comment by yangwao** [comment]
yolo 

Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 7.86 USD/DOT ~ 6.361 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0xd9af0fb0731d82347fdd9553d7bd14ea3a6f54579e85979f43b5d87e2743d418](https://polkadot.subscan.io/extrinsic/0xd9af0fb0731d82347fdd9553d7bd14ea3a6f54579e85979f43b5d87e2743d418)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

File: components/collection/drop/HolderOfGenerative.vue (hunk 1 above)
Comment (yangwao) [reply components/collection/drop/HolderOfGenerative.vue:298]: we can do follow up if so

Comment (Jarsen136) [reply components/collection/drop/HolderOfGenerative.vue:298]: > add this to paid generative also?

Paid generative has handle tx error during the auto-teleport

**Pull request merged**